src/pages/MyKiva/MyKivaPage.vue

```diff
@@ -101,11 +103,12 @@ export default {
 					inviterName: this.userInfo.userAccount?.inviterName ?? null,
 				};
 				this.loans = myKivaQueryResult.my?.loans?.values ?? [];
-				if (bpSidesheetLoan?.lend?.loan) {
-					const bpLoanId = bpSidesheetLoan.lend.loan.id;
-					const filteredLoans = this.loans.filter(loan => loan.id !== bpLoanId);
-					this.loans = [bpSidesheetLoan.lend.loan, ...filteredLoans];
+				this.sidesheetLoan = bpSidesheetLoan?.lend?.loan ?? { id: 0 };
+				const isSideSheetLoanInLoans = this.loans.find(loan => loan?.id === this.sidesheetLoan.id);
+				if (!isSideSheetLoanInLoans && this.sidesheetLoan.id) {
+					this.loans = [this.sidesheetLoan, ...this.loans];
```

Comment:
Won't this get added even if the user didn't lend to it? Is there a simple solution? If not, this might be a reasonable edge case.